ledger/accountdb.go

```diff
@@ -1751,15 +2138,194 @@ func performResourceTableMigration(ctx context.Context, tx *sql.Tx, log func(pro
 	return nil
 }
 
-// removeEmptyAccountData removes empty AccountData msgp-encoded entries from accountbase table
-// and optionally returns list of addresses that were eliminated
-func removeEmptyAccountData(tx *sql.Tx, queryAddresses bool) (num int64, addresses []basics.Address, err error) {
-	if queryAddresses {
-		rows, err := tx.Query("SELECT address FROM accountbase where length(data) = 1 and data = x'80'") // empty AccountData is 0x80
-		if err != nil {
-			return 0, nil, err
-		}
-		defer rows.Close()
+func performTxTailTableMigration(ctx context.Context, tx *sql.Tx, blockDb db.Accessor) (err error) {
+	if tx == nil {
+		return nil
+	}
+
+	dbRound, err := accountsRound(tx)
+	if err != nil {
+		return fmt.Errorf("latest block number cannot be retrieved : %w", err)
+	}
+
+	// load the latest MaxTxnLife rounds in the txtail and store these in the txtail.
+	// when migrating there is only MaxTxnLife blocks in the block DB
+	// since the original txTail.commmittedUpTo preserved only (rnd+1)-MaxTxnLife = 1000 blocks back
```

Comment:
I wonder if now is the time to consider how complicated a db migration will be if we raise MaxTxnLife.  I suspect we will do that soon, as we decrease lambda and it becomes harder to deal with the shorter window for assembling txns from multiple parties in a group.

Reply:
If code can know that a new protocol is going to be enabled, it can start preserving more tail in advance of using that longer tail when the protocol is actually enabled. I think we usually now have upgrade voting completed thousands of rounds before the protocol actually switches so this should be fine?

Reply:
We could probably roll it out in two consecutive consensus upgrades wrapped in one release, like we are doing to go from 1000 blocks to 1001 blocks.